pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -764,8 +764,16 @@ self: super: {
         '';
       };
     in overrideCabal super.servant (old: {
+      # If `enableSeparateDocOutput` is false, `$doc` is not set.
+      # Unfortunately we can't check `enableSeparateDocOutput` directly
+      # here in nix, so we have to check it in bash.
       postInstall = old.postInstall or "" + ''
-        ln -s ${docs} $doc/share/doc/servant
```

Comment:
I think you can use ```''${!outputDoc}``` instead

Reply:
What does `!outputDoc` do? Is that negating a string somehow?

Reply:
@matthewbauer explained me on Github: `${!var}` is bash "variable dereferencing", where `outputDoc` is a variable that's either `out` or `doc`.

So this makes sense.

I've changed it accordingly.